cmd/kube-scheduler/app/server.go

```diff
@@ -323,10 +316,18 @@ func Setup(ctx context.Context, opts *options.Options, outOfTreeRegistryOptions
 		scheduler.WithPodInitialBackoffSeconds(cc.ComponentConfig.PodInitialBackoffSeconds),
 		scheduler.WithExtenders(cc.ComponentConfig.Extenders...),
 		scheduler.WithParallelism(cc.ComponentConfig.Parallelism),
+		scheduler.WithBuildFrameworkCapturer(func(profile kubeschedulerconfig.KubeSchedulerProfile) {
+			// This sets a callback in NewFramework() that will add scheduler profiles to the CC as they are initialized/defaulted
+			// That allows us to log the actual component config and profiles used
+			completedProfiles = append(completedProfiles, profile)
+		}),
 	)
 	if err != nil {
 		return nil, nil, err
 	}
+	if err := options.LogOrWriteConfig(opts.WriteConfigTo, &cc.ComponentConfig, completedProfiles); err != nil {
```

Comment:
why not just override the completed profiles at this point?

Reply:
We could, but since it's only necessary if we're logging/writing the config I figured I'd keep that logic in the function